From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Piotr Szczepanik <piter75@gmail.com>
Date: Sun, 24 Jan 2021 16:14:06 +0100
Subject: [PATCH] add possibility of disabling rk808-rtc

To disable rk808-rtc driver from loading for specific board
add the following stanza to rk808 node in device tree:

	rtc {
		compatible = "rk808-rtc";
		status = "disabled";
	}

This is needed for roc-rk3399-pc plus (a.k.a. Station P1).
Without the change rk808's rtc is initialised and used for time keeping
although there is another rtc (hym8563) that should be actually used.

Signed-off-by: Piotr Szczepanik <piter75@gmail.com>
Signed-off-by: Radxa <dev@radxa.com>
---
 drivers/mfd/mfd-core.c | 2 +-
 drivers/mfd/rk808.c    | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/mfd-core.c b/drivers/mfd/mfd-core.c
index a3a6faa99de0..e48d864110ac 100644
--- a/drivers/mfd/mfd-core.c
+++ b/drivers/mfd/mfd-core.c
@@ -227,7 +227,7 @@ static int mfd_add_device(struct device *parent, int id,
 		}
 
 		if (!pdev->dev.of_node)
-			pr_warn("%s: Failed to locate of_node [id: %d]\n",
+			pr_debug("%s: Failed to locate of_node [id: %d]\n",
 				cell->name, platform_id);
 	}
 
diff --git a/drivers/mfd/rk808.c b/drivers/mfd/rk808.c
index d109b9f14407..d90c45cd5aca 100644
--- a/drivers/mfd/rk808.c
+++ b/drivers/mfd/rk808.c
@@ -145,6 +145,7 @@ static const struct mfd_cell rk808s[] = {
 	{ .name = "rk808-regulator", },
 	{
 		.name = "rk808-rtc",
+		.of_compatible = "rk808-rtc",
 		.num_resources = ARRAY_SIZE(rtc_resources),
 		.resources = rtc_resources,
 	},
-- 
2.25.1

